siriuspy/siriuspy/devices/injsys.py

```diff
@@ -771,3 +773,279 @@ def _command_target(self, cmdtype):
+                self.posang.cmd_update_reference()
+            _time.sleep(InjSysPUModeHandler._DEF_SLEEP)
+            delta = self.posang.delta_angx - self.TS_POSANG_DEFDELTA
+            self.posang.delta_angx = delta
+            if not self._wait(self.posang, 'delta_angx', delta):
+                self._update_status('ERR:Could not do delta AngX.')
+                return False
+
+        if self._abort.is_set():
+            self._clr_flag_abort()
+            return False
+
+        # set DpK trigger event to NLK trigger event
+        dpk_event = self.trignlk.source
+        self.trigdpk.source = dpk_event
+        if not self._wait(self.trigdpk, 'source', dpk_event):
+            self._update_status('ERR:Could not set DpK trigger event.')
+            return False
+        # set DpK trigger DelayRaw to 0
+        self.trigdpk.delay_raw = self.SI_DPKCKR_DLYR_OPT
+        if not self._wait(self.trigdpk, 'delay_raw', self.SI_DPKCKR_DLYR_OPT):
+            self._update_status('ERR:Could not set DpK trigger delay.')
+            return False
+
+        # set DpK Kick
+        self.pudpk.strength = self.SI_DPKCKR_DEFKICK
+        if not self._wait(self.pudpk, 'strength', self.SI_DPKCKR_DEFKICK):
+            self._update_status(
+                'ERR:Could not set DpK Kick to '
+                f'{self.SI_DPKCKR_DEFKICK:.1f}mrad.')
+            return False
+
+        # set pulsed magnet pwrstate and pulse
+        proced = (
+            (self.pudpk.cmd_turn_on, 'turn DpK on.'),
+            (self.punlk.cmd_turn_on, 'turn NLK on.'),
+            (self.pudpk.cmd_turn_on_pulse, 'turn DpK pulse on.'),
+            (self.punlk.cmd_turn_on_pulse, 'turn NLK pulse on.'),
+        )
+        for fun, msg in proced:
+            if self._abort.is_set():
+                self._clr_flag_abort()
+                return False
+            if not fun():
+                self._update_status('ERR:Could not ' + msg)
+                return False
+            _time.sleep(0.5)
+
+        self._update_status('PU mode switched to Optimization!')
+        return True
+
+    @property
+    def is_onaxis(self):
+        """If configuration is OnAxis."""
+        dpk_evt_ok = self.trigdpk.source == self.trignlk.source
+        dpk_dly_ok = self.trigdpk.delay_raw == \
+            self.trignlk.delay_raw + self.SI_DPKCKR_DLYR_ONAXINC
+        dpk_kck_ok = self.pudpk.strength == self.SI_DPKCKR_DEFKICK
+        dpk_on = (self.pudpk.pwrstate == PowerSupplyPU.PWRSTATE.On) and \
+            (self.pudpk.pulse == PowerSupplyPU.PULSTATE.On)
+        nlk_off = (self.punlk.pwrstate == PowerSupplyPU.PWRSTATE.Off) or \
+            (self.punlk.pulse == PowerSupplyPU.PULSTATE.Off)
+        return dpk_evt_ok & dpk_dly_ok & dpk_kck_ok & dpk_on & nlk_off
+
+    def cmd_switch_to_onaxis(self):
+        """Switch to OnAxis."""
+        self._abort.clear()
+        self._update_status('Switching to PU OnAxis config...')
+        if not self.pudpk.interlock_ok:
```

Comment:
these checks for interlocks seem to be the same in every `cmd_switch_to...` method. Maybe create a method like `_check_pu_interlocks` that does this checking.

Reply:
Done!